scylla/src/transport/session.rs

```diff
@@ -83,6 +83,24 @@ use crate::authentication::AuthenticatorProvider;
 #[cfg(feature = "ssl")]
 use openssl::ssl::SslContext;
 
+/**
+ * Translates IP addresses received from ScyllaDB nodes into locally reachable addresses.
+ *
+ * The driver auto-detect new ScyllaDB nodes added to the cluster through server side pushed
+ * notifications and through checking the system tables. For each node, the address the driver
+ * receives corresponds to the address set as `rpc_address` in the node yaml file. In most
+ * cases, this is the correct address to use by the driver and that is what is used by default.
+ * However, sometimes the addresses received through this mechanism will either not be reachable
+ * directly by the driver or should not be the preferred address to use to reach the node (for
+ * instance, the `rpc_address` set on ScyllaDB nodes might be a private IP, but some clients
+ * may have to use a public IP, or pass by a router, e.g. through NAT, to reach that node).
+ * This interface allows to deal with such cases, by allowing to translate an address as sent
+ * by a ScyllaDB node to another address to be used by the driver for connection.
+ *
+ * Please note that the "known nodes" addresses provided while creating the [`Session`]
+ * instance are not translated, only IP address retrieved from or sent by Cassandra nodes
+ * to the driver are.
+ */
```

Comment:
How is this related to slimming down schema agreement API? This should be sent separately.

Reply:
This is a bonus, as described in the PR's description. Opening PRs have nontrivial overhead, so I prefer to add such tiny things to existing ones.

Reply:
> Opening PRs has nontrivial overhead

Is it really that much of an overhead to send a single commit PR? You send it, then somebody reviews it quickly (as it usually is with such simple improvements) and later we can all forget about it before the rest is reviewed.

> I prefer to add such tiny things to existing ones.

No, please, for the sake of order let's not mix up things. Each PR should be concentrated on one topic.